frontend/src/app/pages/settings-page/settings-page.component.ts

```diff
@@ -42,30 +42,49 @@ export class SettingsPageComponent implements OnInit {
             })
     }
 
+    updateUserDetails() {
+        if(!this.validateUserInput()) return;
+        this.usersService.setUserDetails(this.currentUser.id, this.userFirstName.value, this.userLastName.value)
+            .then(() => {
+              this.currentUser.firstName = this.userFirstName.value;
+              this.currentUser.lastName = this.userLastName.value;
+              sessionStorage.setItem('userInfo', JSON.stringify(this.currentUser));
+              this.snackBar.open("User data has been updated.", "Dismiss", {
+                duration: 3000,
+              });
+            });
+    }
+
+    validateUserInput() {
+      if(this.userFirstName.value == "" || this.userLastName.value == "" || this.userEmail.value == "" || (this.userFirstName.value == this.currentUser.firstName && this.userLastName.value == this.currentUser.lastName)) return false;
```

Comment:
Please add some breaklines to this if-statement. It's too long and unreadable :)

```js
if(this.userFirstName.value == "" ||
   this.userLastName.value == "" ||
   ...) {

}
```